Try to warn about all fields having inconsistent settings when they a…

hmusum opened the pull request:
…re used in a field set

Note that this is not done for match settings, as that will change behavior, but we should probably do that change, as it might be wrong now (e.g. you can get 'text' matching when the 2 fields in a fieldset have matching 'exact' and 'word')

@bratseth or @arnej27959 might have opinions about this

hmusum committed:
Try to warn about all fields having inconsistent settings when they are used in a field set

Note that this is not done for match settings, as that will change behavior,
but we should probably do that change, as it might be wrong now
(e.g. you can get 'text' matching when the 2 fields in a fieldset have
matching 'exact' and 'word')

arnej27959 reviewed (commented):
This looks problematic to me.  The new text claims that "matching.getType()" will be used, but the "return" that you want to remove will prevent that from happening.  So we must either do the change or change how we produce the new text.  Also, it doesn't warn about all fields like the commit message would indicate.

arnej27959 committed:
expand warning with information about fields.
make sure that we report back the original fieldset matching
as the effective outcome for now.

hmusum commented:
Thanks, looks good, merging

arnej27959 reviewed (approved):
👍
